### PR TITLE
Add GlobalExceptionHandler and data validation

### PR DESCRIPTION
**Pull request by OlesiaKapustynska**

### CODE REVIEW

File: src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java [hunk 1]
```diff
@@ -1,13 +1,24 @@
...
+import jakarta.validation.constraints.Min;
+import jakarta.validation.constraints.NotBlank;
+import jakarta.validation.constraints.NotNull;
 import java.math.BigDecimal;
 import lombok.Data;
 
 @Data
 public class CreateBookRequestDto {
+    @NotNull
+    @NotBlank
```
Comment (yevheniisynytsia) [inline src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java:10]: `@NotBlank` has a null check inside. `@NotNull` is extra here

**Review by yevheniisynytsia — CHANGES_REQUESTED** [review]
Well done! Left one comment

File: src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java (hunk 1 above)
Comment (OlesiaKapustynska) [reply src/main/java/com/example/bookstore/dto/CreateBookRequestDto.java:10]: fix ✔️

**Review by yevheniisynytsia — APPROVED** [review]
🔥

**Pull request merged**